cdk/cdk_stack.py

```diff
@@ -39,16 +39,24 @@ def __init__(self, scope: cdk.App, id: str, **kwargs) -> None:
 
         apidata_bucket.grant_write(lambda_function)
 
-        api = apigw.LambdaRestApi(
+        second_lambda_function = _lambda.Function(
             self,
-            'WeatherDataAPI',
-            handler=lambda_function,
-            proxy=False
+            'FlattenJsontoCSV',
+            runtime=_lambda.Runtime.PYTHON_3_7,
+            handler='lambda_function.handler',
+            code=_lambda.Code.from_asset('lambda_2'),
+            environment={
+                'BUCKET_NAME': apidata_bucket.bucket_name
+            },
+            timeout=cdk.Duration.minutes(5)
         )
 
-        resource = api.root.add_resource('weather_data')
-        resource.add_method('GET')
+        apidata_bucket.grant_read_write(second_lambda_function)
 
+        s3_notifications.LambdaDestination(second_lambda_function).bind(apidata_bucket).bucket.add_event_notification(
```

Comment:
You shouldn't need to use .bind()...
Can you just do bucket.add_eve...(s3_notifica..) as in the example in doco?

Reply:
yeah seems the document I referred to was outdated, I hope the new one should do it.